### ISSUE TITLE
S3 gateway fails to get object

### ISSUE THREAD

**Issue by donatello** [minio/minio#4243]
Seems to be introduced in recent minio-go update.

Try any get object via s3 gateway - it fails with invalid range offsets error printed on server:

```
ERRO[0003] Unable to write to client.                    cause=Invalid range start and end specified bytes=0-25 source=[gateway-handlers.go:136:gatewayAPIHandlers.GetObjectHandler()] stack=home/aditya/Code/goenv-1.7.5/workspace/src/github.com/minio/minio/cmd/gateway-s3.go:267:(*s3Gateway).GetObject home/aditya/Code/goenv-1.7.5/workspace/src/github.com/minio/minio/cmd/gateway-handlers.go:129:(GatewayLayer).GetObject-fm home/aditya/Code/goenv-1.7.5/workspace/src/github.com/minio/minio/cmd/gateway-handlers.go:135:gatewayAPIHandlers.GetObjectHandler home/aditya/Code/goenv-1.7.5/workspace/src/github.com/minio/minio/cmd/gateway-router.go:81:(gatewayAPIHandlers).GetObjectHandler-fm
```

## Possible Solution

Seems to be a bug in minio-go - will be sending a fix there shortly for review.
## Your Environment

* Version used (`minio version`): Latest master.
* Environment name and version (e.g. nginx 1.9.1):
* Server type and version:
* Operating System and version (`uname -a`):
* Link to your project:

**Comment by deekoder** [comment]
This is a must fix for today's release.

**Comment by donatello** [comment]
This should not be closed until #4256 is merged.
